pkg/scheduler/actions/allocate/allocate.go

```diff
@@ -41,19 +43,24 @@ func (alloc *allocateAction) Execute(ssn *framework.Session) {
 	for !jobsOrderByQueues.IsEmpty() {
 		job := jobsOrderByQueues.PopNextJob()
 		stmt := ssn.Statement()
-		if attemptToAllocateJob(ssn, stmt, job) {
+		alreadyAllocated := podgroup_info.HasTasksAllocated(job)
+		if ok, pipelined := attemptToAllocateJob(ssn, stmt, job); ok {
 			metrics.IncPodgroupScheduledByAction()
 			err := stmt.Commit()
+			if err == nil && !pipelined && !alreadyAllocated {
+				setLastStartTimestamp(job)
```

Comment:
I don't think that this function is the best place for this:
* jobs can be scheduled in other actions (reclaim, preempt, consolidation, maybe something new in the future)
* this function is currently not doing any direct updates on the PodGroupInfo struct, so maybe it will be better to find a place that already does that.

I think that a better place will be in the cache Bind function. I know that it is binding only a specific task out of the whole job, but maybe you can look there at the task that is the job.MinAvailable that is being bound and then know that it is the right moment to add this.
https://github.com/NVIDIA/KAI-Scheduler/blob/main/pkg/scheduler/cache/cache.go#L223

Reply:
You're correct, other actions only virtually allocate jobs. The Allocate action is the only one doing real allocations.
I think it's OK to have this logic in the Allocate action, but maybe move the logic from here to `attemptToAllocateJob` - that way we don't need to add the "pipelined" return value to `attemptToAllocateJob`.

Reply:
@itsomri I wanted the value to be set only if the statement commit succeeds. Should the commit be moved into allocate as well if so?